File: ControlCenter.Web/ClientApp/src/types/serviceArea.ts
import { z } from "zod";

export const UserRoleSchema = z.enum([
  "Ownership",
  "Management",
  "FrontOfficeBdc",
  "InventoryLogistics",
  "Detailing",
  "Photography",
  "TitleAdmin",
]);

export type UserRole = z.infer<typeof UserRoleSchema>;

export const ServiceGroupSchema = z.enum([
  "VehiclePreparation",
  "SalesAndCustomer",
  "FinancialAndCloseout",
  "CommandAndControl",
]);

export type ServiceGroup = z.infer<typeof ServiceGroupSchema>;

export const ServiceAreaCardSchema = z.object({
  id: z.string(),
  title: z.string(),
  description: z.string(),
  destinationPath: z.string(),
  group: ServiceGroupSchema,
  relevantRoles: z.array(UserRoleSchema),
});

export type ServiceAreaCard = z.infer<typeof ServiceAreaCardSchema>;

export const ServiceAreaCardsSchema = z.array(ServiceAreaCardSchema);

export const SERVICE_GROUP_LABELS: Record<ServiceGroup, string> = {
  VehiclePreparation: "Vehicle Preparation",
  SalesAndCustomer: "Sales & Customer",
  FinancialAndCloseout: "Financial & Closeout",
  CommandAndControl: "Command & Control",
};

export const USER_ROLE_LABELS: Record<UserRole, string> = {
  Ownership: "Ownership",
  Management: "Management",
  FrontOfficeBdc: "Front Office / BDC",
  InventoryLogistics: "Inventory / Logistics",
  Detailing: "Detailing",
  Photography: "Photography",
  TitleAdmin: "Title / Admin",
};

export const SERVICE_GROUP_ORDER: ServiceGroup[] = [
  "CommandAndControl",
  "SalesAndCustomer",
  "VehiclePreparation",
  "FinancialAndCloseout",
];

File: ControlCenter.Web/Models/ServiceArea.cs
namespace ControlCenter.Web.Models;

public enum UserRole
{
    Ownership,
    Management,
    FrontOfficeBdc,
    InventoryLogistics,
    Detailing,
    Photography,
    TitleAdmin,
}

public enum ServiceGroup
{
    VehiclePreparation,
    SalesAndCustomer,
    FinancialAndCloseout,
    CommandAndControl,
}

public sealed record ServiceAreaCard
{
    public required string Id { get; init; }
    public required string Title { get; init; }
    public required string Description { get; init; }
    public required string DestinationPath { get; init; }
    public required ServiceGroup Group { get; init; }
    public required IReadOnlyList<UserRole> RelevantRoles { get; init; }
}
